### PR TITLE
GH-830 Teleport to default home feature

### PR DESCRIPTION
**Pull request by igoyek**
## Summary by CodeRabbit

- **New Features**
	- Enhanced home management allowing players with multiple homes to teleport to their default home more efficiently.
	- Updated notifications for players with multiple homes, now including a list of home names.

- **Bug Fixes**
	- Improved logic for retrieving homes to reduce reliance on previous methods, ensuring better performance.

### CODE REVIEW

**Review by Rollczi — CHANGES_REQUESTED** [review]
Ficzer jest już w eternacore

**Review by Rollczi — CHANGES_REQUESTED** [review]
A dobra, już czaję koncept ale boolean w config jest trochę useless. Jak coś jest default to powinno tepać tam i tyle po wpisaniu `/home`

**Review by P1otrulla — APPROVED** [review]

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/command/HomeCommand.java [hunk 1]
```diff
@@ -53,6 +59,17 @@ void execute(@Context Player player) {
...
+                Optional<Home> mainHome = playerHomes.stream()
+                    .filter(home -> home.getName().equals(this.homesConfig.defaultHomeName))
+                    .findFirst();
+
+                if (mainHome.isPresent()) {
+                    this.homeTeleportService.teleport(player, mainHome.get());
+                    return;
+                }
+            }
+
```
Comment (CitralFlo) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/home/command/HomeCommand.java:72]: please remove Boolean form config and make use of existing firstHome value if possible simplyfing the code

**Review by CitralFlo — CHANGES_REQUESTED** [review]

**Review by vLuckyyy — APPROVED** [review]
If you have tested, everything in my opinion is good

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/command/HomeCommand.java [hunk 2]
```diff
@@ -46,13 +52,24 @@ void execute(@Context Player player) {
             return;
         }
 
-        if (playerHomes.size() != 1) {
+        Home firstHome = playerHomes.iterator().next();
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/home/command/HomeCommand.java:55]: Why you move this variable to the beginning?

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/command/HomeCommand.java [hunk 3]
```diff
@@ -46,13 +52,24 @@ void execute(@Context Player player) {
...
+
+        if (playerHomes.size() > 1) {
             String homes = String.join(
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/home/command/HomeCommand.java:58]: Move it after related if condition

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/command/HomeCommand.java [hunk 4]
```diff
@@ -46,13 +52,24 @@ void execute(@Context Player player) {
...
                     .toList());
 
+            Optional<Home> mainHome = playerHomes.stream()
+                .filter(home -> home.getName().equals(this.homesConfig.defaultHomeName))
+                .findFirst();
+
+            if (mainHome.isPresent()) {
+                this.homeTeleportService.teleport(player, mainHome.get());
+                return;
+            }
```
[the remark this answers is not in the document]
Comment (Rollczi) [reply eternalcore-core/src/main/java/com/eternalcode/core/feature/home/command/HomeCommand.java:71]: dupa

**Review by CitralFlo — APPROVED** [review]
LGTM

**Review by Rollczi — APPROVED** [review]

**Pull request merged**